### PR TITLE
legacy: OaiHarvest removal

### LINKED ISSUE #2936: OAIHarvest: stuck when trying to harvest
For some estrange reason after deploying 3d60603e08d0a4e95e79d70d37bf9107531be30a on CDS, specially the modification done on `modules/oaiharvest/lib/oai_harvest_getter.py`, OAIHarvest gets stuck on `source arXiv is going to be updated`.

There is no error on the logs or any kind of output.

cc @aw-bib

### PR DESCRIPTION
**Pull request by jalavik**
Removal of legacy OAIHarvest module. I moved some required parts to the oaiharvester module. I will remove that module as work on `invenio-oaiharvester` progresses (within a week or so).

Also added some minor new commits touching these files (e.g. the plotextractor part).

Supersedes #3298 and closes #2936.

@jirikuncar

### CODE REVIEW

File: invenio/modules/oaiharvester/config.py [hunk 1]
```diff
@@ -19,8 +19,20 @@
...
+import os
+import sys
+
+
 OAIHARVESTER_DEFAULT_NAMESPACE_MAP = {
     None: "http://www.openarchives.org/OAI/2.0/",
 }
+"""The default namespace used when handling OAI-PMH results."""
+
+OAIHARVESTER_STORAGEDIR = os.path.join(sys.prefix, "var/data/oaiharvester/storage")
```
Comment (jirikuncar) [inline invenio/modules/oaiharvester/config.py]: please use `CFG_DATADIR` and `os.path.join`

File: invenio/modules/oaiharvester/errors.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# -*- coding: utf-8 -*-
+# -*- mode: python; coding: utf-8; -*-
```
Comment (jirikuncar) [inline invenio/modules/oaiharvester/errors.py]: why do you need `mode: python;`?

Comment (jalavik) [inline invenio/modules/oaiharvester/errors.py]: Hmm, did not add it myself. Must be from the legacy files. I will kill it.

**Comment by jirikuncar** [comment]
@jalavik shall I already prepare `invenio-oaiharvester` repo? Which files do you want to include there?

**Comment by jalavik** [comment]
> shall I already prepare invenio-oaiharvester repo?

Sure, that would be great. The files to include would be the oaiharvester module excluding the following directories and files:

* `static/`
* `views.py`

I guess the rest can go as a start and I will rewrite them heavily.

PS: I will rephrase as asked.

**Comment by jalavik** [comment]
Rephrased and added f056534

**Pull request merged**